### PR TITLE
Migrate to tf sdk

### PR DESCRIPTION
**Pull request by jeffb4**
This changeset encapsulates the process of migrating to the new terraform plugin sdk

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,14 +3,11 @@ module github.com/terraform-providers/terraform-provider-rancher2
...
-	github.com/gophercloud/gophercloud v0.1.0 // indirect
 	github.com/hashicorp/go-version v1.2.0
-	github.com/hashicorp/terraform v0.12.0
-	github.com/peterbourgon/diskv v2.0.1+incompatible // indirect
+	github.com/hashicorp/terraform-plugin-sdk v1.0.0
 	github.com/rancher/norman v0.0.0-20191003174345-0ac7dd6ccb36
-	github.com/rancher/types v0.0.0-20191003184925-ab3403a2c8ca
-	golang.org/x/crypto v0.0.0-20190611184440-5c40567a22f8
-	golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45 // indirect
+	github.com/rancher/types v0.0.0-20191017022642-c4c5ca1581f9
```
Comment (rawmind0) [inline go.mod]: Could you please update rancher/types to version used at Rancher v2.3.2?? :)

```
-	github.com/rancher/types v0.0.0-20191017022642-c4c5ca1581f9
+	github.com/rancher/types v0.0.0-20191023170600-1563e7864c02
```

**Review by rawmind0 — CHANGES_REQUESTED** [review]
Thanks for the PR @jeffb4 .

It seems that not all vendor files update were committed. I've got some updates executing `GO111MODULE=on go mod vendor` from your PR. 

Could you please update `rancher/types` version as requested, update vendor files and commit??

File: go.mod (hunk 1 above)
Comment (jeffb4) [reply go.mod]: Updated and pushed :)

**Review by rawmind0 — COMMENTED** [review]
@jeffb4 , rancher/types is updated correctly, but getting new vendor files when execute `GO111MODULE=on go mod vendor` from your PR. Could you please take a look??

```
# git status
On branch 168
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   go.mod
	modified:   vendor/modules.txt

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	vendor/github.com/apparentlymart/go-textseg/textseg/make_tables.go
	vendor/github.com/apparentlymart/go-textseg/textseg/make_test_tables.go
	vendor/github.com/hashicorp/hcl2/hcl/hclsyntax/expression_vars_gen.go
	vendor/github.com/ulikunitz/xz/example.go
	vendor/golang.org/x/sys/unix/mkasm_darwin.go
	vendor/golang.org/x/sys/unix/mkpost.go
	vendor/golang.org/x/sys/unix/mksyscall.go
	vendor/golang.org/x/sys/unix/mksyscall_aix_ppc.go
	vendor/golang.org/x/sys/unix/mksyscall_aix_ppc64.go
	vendor/golang.org/x/sys/unix/mksyscall_solaris.go
	vendor/golang.org/x/sys/unix/mksysctl_openbsd.go
	vendor/golang.org/x/sys/unix/mksysnum.go
	vendor/golang.org/x/sys/unix/types_aix.go
	vendor/golang.org/x/sys/unix/types_darwin.go
	vendor/golang.org/x/sys/unix/types_dragonfly.go
	vendor/golang.org/x/sys/unix/types_freebsd.go
	vendor/golang.org/x/sys/unix/types_netbsd.go
	vendor/golang.org/x/sys/unix/types_openbsd.go
	vendor/golang.org/x/sys/unix/types_solaris.go
	vendor/golang.org/x/text/unicode/bidi/gen.go
	vendor/golang.org/x/text/unicode/bidi/gen_ranges.go
	vendor/golang.org/x/text/unicode/bidi/gen_trieval.go
	vendor/golang.org/x/text/unicode/norm/maketables.go
	vendor/golang.org/x/text/unicode/norm/triegen.go
```

**Comment by jeffb4** [comment]
I wonder if the difference is that I'm running go mod vendor on a Mac, and maybe you're in Linux? If that's the case you're more than welcome to commit/push those changes to my fork branch

**Comment by jeffb4** [comment]
Difference was that I was using go1.13 instead of go1.12 (as specified by go.mod). Downgraded, revendored, committed, pushed

**Review by rawmind0 — APPROVED** [review]
LGTM

**Pull request merged**